internal/datadog/metrics_client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadog // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog"
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type metricsClient struct {
+	meter  metric.Meter
+	gauges map[string]float64
+	mutex  sync.Mutex
+}
+
+// NewMetricClient returns a new metrics client
+func NewMetricClient(meter metric.Meter) metrics.StatsClient {
+	// We need to lock the mutex here to avoid
+	// duplicate metrics being created by both datadogconnector and exporter
+	var mutex sync.Mutex
+	mutex.Lock()
+	defer mutex.Unlock()
```

Comment:
I had to change this to move the mutex variable to the package level instead of in the constructor. `NewMetricsClient` is called by both the connector and the exporter because if a customer is using a gateway deployment it's possible that they're running the Collector with the connector and no exporter.

It's ok for `metrics.Client` to be overridden when both the connector and exporter are creating new metrics clients, but the mutex is just to ensure that each metrics client is created correctly.